vendor/bat-native-ledger/include/bat/ledger/ledger.h

```diff
@@ -214,8 +216,8 @@ class LEDGER_EXPORT Ledger {
       int year,
       ledger::GetBalanceReportCallback callback) const = 0;
 
-  virtual std::map<std::string, ledger::BalanceReportInfoPtr>
-  GetAllBalanceReports() const = 0;
+  virtual void GetAllBalanceReports(
+      ledger::GetBalanceReportListCallback callback) const = 0;
```

Comment:
This change will require a change in `BATBraveLedger.mm` for iOS (CI will fail until its fixed). I will try and find some time to add the change today

Reply:
OK, thanks Kyle.

Reply:
@emerick Ah just realized this is `GetAllBalanceReports`, not `GetBalanceReport`. 

The latter needs a very minor fix because iOS was relying on the original implementation of this being a non-escaping callback, but I can push that change now and create an issue in brave-ios for the callsite to adjust accordingly